Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs

```diff
@@ -206,29 +206,11 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
             var pointers = RequestPointers(controllerType, controllingHand);
             var inputSource = InputSystem?.RequestNewGenericInputSource($"{CurrentControllerType} Controller {controllingHand}", pointers);
 
-            GenericOpenVRController detectedController = null;
-
-            switch (CurrentControllerType)
-            {
-                case SupportedControllerType.GenericOpenVR:
-                    detectedController = new GenericOpenVRController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.ViveWand:
-                    detectedController = new ViveWandController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.ViveKnuckles:
-                    detectedController = new ViveKnucklesController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.OculusTouch:
-                    detectedController = new OculusTouchController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.OculusRemote:
-                    detectedController = new OculusRemoteController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-            }
+            GenericOpenVRController detectedController = Activator.CreateInstance(controllerType, 
+                new object[] { TrackingState.NotTracked, controllingHand, inputSource }) as GenericOpenVRController;
```

Comment:
Does this still allow OpenVR apps to work with the Windows Mixed Reality controllers when using the WMR driver for Steam?

Reply:
I'll pull and check now.

Reply:
You can simplify this whole line down to:
```


var detectedController = Activator.CreateInstance(controllerType, TrackingState.NotTracked, controllingHand, inputSource, null) as GenericOpenVRController;
```